pkgs/build-support/go/README.md

```diff
@@ -0,0 +1,42 @@
+## Go toolchain/builder upgrade policy
+
+Go promises that "programs written to the Go 1 specification will continue to compile and run correctly, unchanged, over the lifetime of that specification" [1].
+Newer toolchain versions should build projects developed against older toolchains without problems.
+
+There are however Go packages depending on internal APIs of the toolchain/runtime/stdlib that is not covered by the Go compatibility promise.
+These packages may break on toolchain updates.
+We name the set of these packages `toolchain-breaking`.
+
+There is another set of packages that depends on the toolchain, but in another way:
+Packages providing development support for the Go language (like `gopls`, `golangci-lint`,...) must be compiled with the version they should be used for.
+If `gopls` is compiled for Go 1.23, it won't work for projects that require Go 1.24.
+We name the set of these packages `toolchain-latest`.
+
+Go only ever has two supported toolchains. With a new minor release, the second last Go toolchain is automatically end of life, meaning it won't receive security updates anymore.
+
+Based on this, we align on the following policy for toolchain/builder upgrades:
+
+1. Toolchain (the `go` package) and builder (`buildGoModule`) are upgraded to the latest minor release of Go as soon as it is released.
+
+2. Packages in `toolchain-latest` SHOULD NOT pin a toolchain version.
```

Comment:
We just need to make sure to also update them on time. golangci-lint usually needs a fix for the latest go version. Maybe we should update them in tandem or bump go to the latest when we have that update?

Reply:
Hm, I see your point, but how do we coordinate that? I think from a Go teams perspective, the package maintainers are responsible to update in time.

Does golangci-lint break if compiled for with a newer Go toolchan before the updated? Or does it just not provide the needed support for that toolchain?

Reply:
The bump will also go through staging, so there should usually be some time for package maintainers to update before the bump reaches unstable.

Reply:
> how do we coordinate that? I think from a Go teams perspective, the package maintainers are responsible to update in time.

The go team just bumps golangci-lint to the latest version if it didn't happen when updating the go version.



> Does golangci-lint break if compiled for with a newer Go toolchain before the updated? Or does it just not provide the needed support for that language version?

The answer is: depends but they don't support it https://github.com/golangci/golangci-lint/issues/5225